packages/edit-site/src/components/global-styles/global-styles-provider.js

```diff
@@ -30,6 +31,37 @@ export function mergeBaseAndUserConfigs( base, user ) {
 	} );
 }
 
+function resolveBlockStyleVariations( userConfig ) {
+	const sharedVariations = userConfig.styles?.blocks?.variations;
+
+	if ( ! sharedVariations ) {
+		return userConfig;
+	}
+
+	const variationsConfig = JSON.parse( JSON.stringify( userConfig ) );
```

Comment:
Ah nice, thanks for the heads up. This code has been in play since last year before that util was created by the looks of it. I'll get it updated 👍